packages/react/src/popover/root/PopoverRoot.tsx

```diff
@@ -170,7 +171,7 @@ function PopoverRootComponent({ props }: { props: PopoverRoot.Props }) {
       setMounted,
       transitionStatus,
       triggerElement,
-      setTriggerElement,
+      setTriggerElement: props.triggerRef ? () => {} : setInternalTriggerElement,
```

Comment:
Also, there might be a conflict if you try both passing the `triggerRef` and rendering the `Popover.Trigger` within. I understand that it's an edge case and not the intended way to use the component. But maybe there's a way to be more defensive against that 🤔.

I believe that instead of using a noop for the `setTriggerElement`, some form of a warning message can be provided:
```suggestion
      setTriggerElement: props.triggerRef 
        ? () => { console.warn("WHAT THE HELL ARE YOU DOING? THERE'S A REF ALREADY!") } 
        : setInternalTriggerElement,
```